### PR TITLE
Create and use py3_compat.makedirs(exist_ok)

### PR DESCRIPTION
**Pull request by mppf**
We have seen nightly testing error where checking if a directory existed or not and then creating it if not resulted in errors for the directory already existing. To resolve these, add a `makedirs` function to `py3_compat` that accepts an `exist_ok` argument. Then, update `os.makedirs` calls used in testing to call the new function with `exist_ok=True`. (The exist_ok argument is already available in `os.makedirs` in Python 3 so one day the py3_compat function can be removed).

- [x] hellos pass with Python 2
- [x] hellos pass with Python 3
- [x] examples/benchmarks/shootout performance tests complete with Python 2
- [x] full local testing with paratest

Reviewed by @ronawho - thanks!

### CODE REVIEW

**Comment by mppf** [comment]
@ronawho - could you have a look & suggest how to test this? Thanks

**Comment by ronawho** [comment]
I don't think there's a concurrency race here, but I suppose if there's a glitch in NFS the "exists" could return false even if the dir exists and then come back online by the time we try to create it. We're bound to run into lots of issues if that happens, but I'm not opposed to fixing this one. 

There's probably other makedirs instances that should be updated. It' might just be worth adding a wrapper to our py3_compat.py shim like:
```python
def makedirs(path, mode=0o777, exist_ok=False):
    if sys.version_info[0] >= 3:
        os.makedirs(path, mode, exist_ok)
    else:
        try:
            os.makedirs(path, mode)
        except OSError as e:
            if not exist_ok or e.errno != errno.EEXIST:
                raise
```

https://github.com/python/cpython/blob/3.7/Lib/os.py#L196

**Comment by mppf** [comment]
@ronawho - I did that. Please have a look & advise on how you'd recommend testing this, thanks.

**Comment by ronawho** [comment]
> Please have a look & advise on how you'd recommend testing this, thanks.

I have no idea how you'd test this. The execution limiter is the only place that's designed to be run concurrently.

**Comment by mppf** [comment]
@ronawho - I think this is ready for a final look. If it looks OK with you (and the testing I did seems reasonable for the moment) then I will merge it. Thanks.

File: util/start_test [hunk 1]
```diff
@@ -615,7 +616,7 @@ def check_environment_with_args():
     global logs_dir
     logs_dir = os.path.join(test_dir, "Logs")
     if not os.path.isdir(logs_dir):
```
Comment (ronawho) [inline util/start_test]: we could probably remove all the "if dir does not exist" checks now, but I don't mind leaving them

**Pull request merged**